docs/headless-js-android.md

```diff
@@ -183,13 +277,124 @@ public class NetworkChangeReceiver extends BroadcastReceiver {
 
     public static boolean isNetworkAvailable(Context context) {
         ConnectivityManager cm = (ConnectivityManager)
-        context.getSystemService(Context.CONNECTIVITY_SERVICE);
+                context.getSystemService(Context.CONNECTIVITY_SERVICE);
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            Network networkCapabilities = cm.getActiveNetwork();
+
+            if(networkCapabilities == null) {
+                return false;
+            }
+
+            NetworkCapabilities actNw = cm.getNetworkCapabilities(networkCapabilities);
+
+            if(actNw == null) {
+                return false;
+            }
+
+            if(actNw.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) || actNw.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR) || actNw.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET)) {
+                return true;
+            }
+
+            return false;
+        }
```

Comment:
Why is this needed?

Reply:
`getActiveNetworkInfo` is deprecated,  so we can use it in all APIs below 29.

https://developer.android.com/reference/android/net/NetworkInfo

One of the proposed on the docs is to use `getNetworkCapabilities`

https://developer.android.com/reference/android/net/ConnectivityManager#getNetworkCapabilities(android.net.Network)